osgar/lib/test_quaternion.py

```diff
@@ -24,23 +24,16 @@ def test_rotate_z90(self):
         q = [ 0, 0, 0.7071068, 0.7071068 ]
         actual = quaternion.rotate_vector(v, q)
         expected = [0, 1, 0]
-        self.assertAlmostEqual(expected[0], actual[0], places=6)
-        self.assertAlmostEqual(expected[1], actual[1], places=6)
-        self.assertAlmostEqual(expected[2], actual[2], places=6)
+        self.assertXYZEqual(expected, actual)
 
         actual = quaternion.rotate_vector(expected, q)
         expected = [-1, 0, 0]
-        self.assertAlmostEqual(expected[0], actual[0], places=6)
-        self.assertAlmostEqual(expected[1], actual[1], places=6)
-        self.assertAlmostEqual(expected[2], actual[2], places=6)
+        self.assertXYZEqual(expected, actual)
 
     def test_from_axis_angle(self):
         actual = quaternion.from_axis_angle((0,0,1), math.radians(90))
         expected = [0, 0, 0.7071068, 0.7071068]
-        self.assertAlmostEqual(expected[0], actual[0], places=6)
-        self.assertAlmostEqual(expected[1], actual[1], places=6)
-        self.assertAlmostEqual(expected[2], actual[2], places=6)
-        self.assertAlmostEqual(expected[3], actual[3], places=6)
+        self.assertQuaternionEqual(expected, actual)
```

Comment:
I would use `assertArrayAlmostEqual()`, which would check for the tupple size (i.e. not separate method for 3 and for 4 dimensions)??

Reply:
That's the question I was asking - is it better being specific or generic... So you are for the generic approach, ok.

But Array is not used by anyone - unittest uses `List`, `Tuple` or `Sequence` (see https://github.com/python/cpython/blob/master/Lib/unittest/case.py).

Reply:
Please note also my initial question:
> Using the generic approach would allow me to also replace angles compare and matrix compare. However I was afraid that the generic approach would not offer errors/messages actionable enough. Opinions welcome.

Using one common/generic routine gives us also a generic output when test fails.

Reply:
See this approach in the latest commits.